### ISSUE TITLE
pnet_datalink: invalid CString pointer use in unix_interfaces::interfaces ?

### ISSUE THREAD

**Issue by decathorpe** [libpnet/libpnet#474]
```
warning: getting the inner pointer of a temporary `CString`
  --> src/unix_interfaces.rs:79:62
   |
79 |             iface.index = libc::if_nametoindex(name.unwrap().as_ptr());
   |                                                ------------- ^^^^^^ this pointer will be invalid
   |                                                |
   |                                                this `CString` is deallocated at the end of the statement, bind it to a variable to extend its lifetime
   |
   = note: `#[warn(temporary_cstring_as_ptr)]` on by default
   = note: pointers do not have a lifetime; when calling `as_ptr` the `CString` will be deallocated at the end of the statement because nothing is referencing it as far as the type system is concerned
   = help: for more information, see https://doc.rust-lang.org/reference/destructors.html
```

Looking at the code, it looks like this might be a false positive warning, because the pointer is only used to calculate the `index` field, and not used for anything else. If that is the case, could `#[allow(temporary_cstring_as_ptr)]` be added to this block / statement so users are not getting this "concerning" warning?

**Comment by mrmonday** [comment]
This is valid - pushed a fix in commit d49605d.

**Comment by mrmonday** [comment]
Thanks for the report!
